channel-sdk/src/main/java/com/zello/channel/sdk/platform/Audio.kt

```diff
@@ -128,18 +128,19 @@ internal class Audio private constructor(context: Context, val logger: SessionLo
 
 	companion object {
 
+		// FIXME: Make this some kind of language-appropriate late-initialized property to clean up casting in getter
 		private var _manager: Audio? = null
 
-		@JvmStatic operator fun get(context: Context, logger: SessionLogger?): Audio? {
+		@JvmStatic operator fun get(context: Context, logger: SessionLogger?): Audio {
 			if (_manager != null) {
-				return _manager;
+				return _manager as Audio
 			}
 			synchronized(Audio::class.java) {
 				if (_manager == null) {
 					_manager = Audio(context.applicationContext, logger)
 				}
 			}
-			return _manager
+			return _manager as Audio
```

Comment:
I think you should be able to do something like this:
```
return _manager ?: Audio(context.applicationContext, logger)
```
Then you wouldn't need the casting. You would probably have to extend the synchronous block to wrap the whole thing, since the whole thing would now be one line. 

Granted I'm not understanding the broader context of the synchronization, so I don't know if that messes with some consideration there

Reply:
I think the synchronization is to ensure that if multiple threads attempt to get the `Audio` object for the first time, at the same time, that it will only be initialized once. I *suspect* there may be a way to do that automatically, but I could be wrong. My Swift knowledge really interferes with my Kotlin understanding.

Reply:
Kotlin `object`s have that protection, but unfortunately they can't take constructor parameters; since you need parameters here, what you've done is a perfectly valid approach.

If that's all the sync is for, then I'd think you should be fine just synchronizing the whole block. It might be marginally less performant, but unless there were a ton of threads requesting the manager at once I can't imagine the impact would be noticeable.

If you are _really_ concerned, though, you could also do something like this:
```
_manager?.let { return it }
synchronized(Audio::class.java) {
    val manager = _manager ?: Audio(context.applicationContext, logger)
    _manager = manager
    return manager
}
```

The `?.let` just creates a local `val` within that block so that it is guaranteed to be non-null